server/channels/api4/post.go

```diff
@@ -651,7 +651,8 @@ func getPostThread(c *Context, w http.ResponseWriter, r *http.Request) {
 	fromPost := r.URL.Query().Get("fromPost")
 	// Either only fromCreateAt must be set, or both fromPost and fromCreateAt must be set
 	if fromPost != "" && fromCreateAt == 0 {
-		c.SetInvalidParam("if fromPost is set, then fromCreatAt must also be set")
+		c.SetInvalidParam("if fromPost is set, then fromCreateAt must also be set")
+		return
```

Comment:
Previously, if this case was hit, the HTTP response was (correctly) a 400. After the library migration to `github.com/klauspost/compress/gzhttp`, the response is (incorrectly) a 200.

Without this fix, the changes in the PR broke a test. I was not able to track down why the behaviour changed. But the code here was surely wrong and hence I fixed it has part of this PR.

Reply:
You are the reviewer now: https://github.com/mattermost/mattermost/pull/24288 :)